lib/vagrant-libvirt/action.rb

```diff
@@ -317,6 +317,58 @@ def self.action_ssh_run
         end
       end
 
+      def self.action_snapshot_delete
+        Vagrant::Action::Builder.new.tap do |b|
+          b.use ConfigValidate
+          b.use Call, IsCreated do |env, b2|
+            if env[:result]
+              b2.use SnapshotDelete
+              next
+            else
+              b2.use MessageNotCreated
+            end
+          end
+        end
+      end
+
+      # This is the action that is primarily responsible for restoring a snapshot
+      def self.action_snapshot_restore
+        Vagrant::Action::Builder.new.tap do |b|
+          b.use ConfigValidate
+          b.use Call, IsCreated do |env, b2|
+            if env[:result]
+              b2.use SnapshotRestore
+              next
+            else
+              b2.use MessageNotCreated
+            end
+
+            b2.use Call, IsRunning do |env2, b3|
+              unless env2[:result]
+                b3.use MessageNotRunning
+                next
+              end
+              b3.use SnapshotDelete
```

Comment:
Is this intended to automatically delete snapshots associated with machines already deleted during the restore?